impacket/examples/mssqlshell.py

```diff
@@ -230,6 +231,7 @@ def do_enum_impersonate(self, line):
             self.sql.printReplies()
             self.sql.rows = result
             self.sql.printRows()
+            self.sql_query("use " + old_db)
         except:
             pass

```

Comment:
Setting original DB as you fixed
```suggestion
            try:
                self.sql_query("use " + old_db)
            except:
                pass

```